### PR TITLE
Setting kerberos user store domains.

### PR DESCRIPTION
**Pull request by omindu**

### CODE REVIEW

File: src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java [hunk 1]
```diff
@@ -190,6 +193,9 @@ else if (KerberosGrantConstants.KERBEROS_IDP_SPNPASSWORD.equals(property.getName
                         kerberosUsersId.length();
                 AuthenticatedUser kerberosUser = new AuthenticatedUser();
                 kerberosUser.setUserName(kerberosUsersId.substring(0, indexOfAt));
+                if(StringUtils.isNotEmpty(userstoreDomains)){
```
Comment (mefarazath) [inline src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java]: unless we iterate the user stores and make sure that this use is indeed present in the defined userstore domain can we set this as user store domain?

Check https://github.com/wso2-extensions/identity-local-auth-iwa-kerberos/blob/master/components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java#L241 we do this check.

File: src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java [hunk 2]
```diff
@@ -189,7 +206,20 @@ else if (KerberosGrantConstants.KERBEROS_IDP_SPNPASSWORD.equals(property.getName
...
+
+                if (StringUtils.isNotEmpty(userstoreDomains)) {
+                    String[] userStores = userstoreDomains.split(",");
+
+                    for (String userStoreDomain : userStores) {
+                        if (isUserExistsInUserStore(kerberosUser.getUserName(), tenantDomain, userStoreDomain)) {
+                            kerberosUser.setUserStoreDomain(userStoreDomain);
+                            break;
+                        }
+                    }
```
Comment (mefarazath) [inline src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java:225]: What happens when the user doesn't exist in any of the specified domain?
Shouldn't we fail the request?

Comment (omindu) [reply src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java:225]: So we have 2 scenarios. One where the IDP configs doesn't have user store domains and the other one where the user does not exist in any of the configured user stores.

For the first case, we will treat it as a federated scenario and fail grant validation for the second case.

Comment (omindu) [reply src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java:225]: Fixed

**Review by IsuraD — APPROVED** [review]

**Pull request merged**